packages/server/lib/factory/loggerFactory.js

```diff
@@ -1,239 +1,39 @@
-    case 'debug':
-      return chalk.bold.green(`${level}: `);
-    case 'silly':
-    default:
-      return chalk.bold.gray(`${level}: `);
-  }
-}
-class ConsoleAsync extends TransportStream {
-  constructor(options = {}) {
-    super(options);
-
-    this.name = options.name || 'console-async';
-    this.rootDir = options.rootDir;
-  }
-
-  log(info, callback) {
-    this._log(info, callback);
-  }
-
-  async _log(meta, callback) {
-    let message = meta.message;
-
-    // Parse error
-    if (meta?.error?.message) {
-      try {
-        const parsedErrorData = await parseError(
-          meta.error,
-          resolveErrorType(meta.error)
-        );
-        message = await devFormatError(parsedErrorData, this.rootDir);
-      } catch {
-        // Fallback to original error messsage
-        message = meta?.error?.message;
-      }
-    }
-
-    (console[meta.level] ?? console.log)(
-      `${colorizeLevel(meta.level)}${message}`
-    );
-
-    // Log additional error params
-    if (meta?.error?.getParams) {
-      (console[meta.level] ?? console.log)(chalk.redBright(`Error Params:\n`));
-      (console[meta.level] ?? console.log)(meta?.error?.getParams());
-    }
-
-    callback();
-  }
-}
-
-const devLogger = createLogger({
-  format: json(),
-  levels: config.npm.levels,
-  transports: [
-    new ConsoleAsync({
-      rootDir: process.cwd(),
-    }),
-  ],
-});
-
-function formatMetaSimple(meta) {
-  let keys = Object.keys(meta).filter(
-    key => ['level', 'timestamp', 'message'].indexOf(key) === -1
-  );
-  if (!meta || !keys.length) {
-    return '';
-  }
-
-  let lines = keys.map(key => {
-    let value = meta[key];
-    if (value instanceof Error) {
-      return key + ': ' + indentLines(value.stack, '   ', true);
-    } else if (value instanceof Object) {
-      return (
-        key + ': ' + indentLines(JSON.stringify(value, null, 4), '   ', true)
-      );
-    } else {
-      return key + ': ' + value;
-    }
-  });
-
-  return '\n - ' + lines.join('\n - ');
-}
-
-function indentLines(string, spaces, skipFirstLine) {
-  let lines = string.split('\n');
-
-  let indentedLines = lines.map((line, index) => {
-    if (!index && skipFirstLine) {
-      return line;
-    }
-
-    return spaces + line;
+/**
+ * Logger factory for IMA.js server
+ * Provides JSON-based logging for production environments
+ */
+
+/**
+ * Private method to write log messages with JSON formatting
+ * @param {string} level - Log level (log, info, warn, error)
+ * @param {any[]} data - data to log
+ */
+function _writeLog(level, data) {
+  const logData = JSON.stringify({
+    level,
+    data: typeof data === 'string' ? { message: data } : data,
   });
 
-  return indentedLines.join('\n');
-}
-
-function formatMetaJSON(meta) {
-  let keys = Object.keys(meta).filter(
-    key => ['level', 'timestamp', 'message'].indexOf(key) === -1
-  );
-  if (!meta || !keys.length) {
-    return '';
+  if (level === 'error') {
```

Comment:
`trace` also prints to `stderr`, maybe it would be better to pass `process.stdout` / `process.stderr` as a parameter to `_writeLog` instead of this `if`/`else`?